allure-java-commons/src/main/java/io/qameta/allure/util/ResultsUtils.java

```diff
@@ -225,9 +225,10 @@ public static Link createLink(final io.qameta.allure.TmsLink link) {
     @SuppressWarnings("PMD.UseObjectForClearerAPI")
     public static Link createLink(final String value, final String name,
                                   final String url, final String type) {
-        final String resolvedName = firstNonEmpty(value).orElse(name);
+        final String resolvedName = firstNonEmpty(name).orElse(value);
+        final String resolvedValue = firstNonEmpty(value).orElse(name);
         final String resolvedUrl = firstNonEmpty(url)
-                .orElseGet(() -> getLinkUrl(resolvedName, type));
+                .orElseGet(() -> getLinkUrl(resolvedValue, type));
```

Comment:
currently `value` is alias to `name`, I'm ok with changing priority (to use `name` if specified). But you also add logic that `value` will be used to create link url even if name is specified. Thats sounds bad as increases complexity

Reply:
at the moment `url` and `name` (in Link class, few lines below, the `Link` in return statement) will use the same variable, either `value` or `name`, just one of them to create both `URL` and it's `name`, which will usually be just some number like this [369](https://github.com/allure-framework/allure-java/pull/396)

Should there be a way to create `Link` with `URL` and some other `name`? Like:
[A long text, that does no have 369 number in it](https://github.com/allure-framework/allure-java/pull/396)
something like `@Issue(value="369" name="The text")`

Please, don't get me wrong, i am not trying to force this change on Allure. I am just asking if my intuition is right or wrong.